### PR TITLE
Flatpak: Add `org.freedesktop.Platform.ffmpeg-full` Extension Point

### LINKED ISSUE #281: Cannot play most videos due to lack of codecs
### What Happened?

After elementary OS 6 October updates, Videos is installed as a Flatpak and can no longer play videos, like the mp4 screencast I tested it with. 

When run with the Terminal, Videos throws this error: (io.elementary.videos:2): Gtk-WARNING **: 11:19:37.121: Negative content width -11 (allocation 1, extents 6x6) while allocating gadget (node actionbar, owner GtkActionBar). 

The same video can be played by Clapper (Gtk4, Flatpak) without problem. I have installed the following runtimes: org.freedesktop.Platform.ffmpeg-full org.freedesktop.Platform.Gstreamer.gstreaner-vaapi org.freedesktop.Platform.openh264

### Steps to Reproduce

1. Install io.elementary.videos 2.8.0 branch stable from repo appcenter (system-wide)
2. Open an mp4 video

### Expected Behavior

Videos is able to play videos

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

```shell
(io.elementary.videos:2): Gtk-WARNING **: 11:19:37.121: Negative content width -11 (allocation 1, extents 6x6) while allocating gadget (node actionbar, owner GtkActionBar)
```

### Hardware Info

CPU: Intel Core i7-9700

GPU: Intel UHD 630

`flatpak run org.freedesktop.Platform.VaInfo`: 
libva info: VA-API version 1.12.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/intel-vaapi-driver/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_12
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.12 (libva 2.9.1)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 21.3.3 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileNone                   :	VAEntrypointStats
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Simple            :	VAEntrypointEncSlice
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointFEI
      VAProfileH264Main               :	VAEntrypointEncSliceLP
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointFEI
      VAProfileH264High               :	VAEntrypointEncSliceLP
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264ConstrainedBaseline:	VAEntrypointFEI
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          :	VAEntrypointVLD
      VAProfileVP8Version0_3          :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointFEI
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointEncSlice
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointVLD

### PR DESCRIPTION
**Pull request by Marukesu**
Allows videos to use the codecs provided by the extension when installed.

Should fix #281.

### CODE REVIEW

File: io.elementary.videos.yml [hunk 1]
```diff
@@ -17,9 +17,18 @@ finish-args:
   - '--talk-name=org.gnome.SettingsDaemon'
 
   - '--metadata=X-DConf=migrate-path=/io/elementary/videos/'
+
+add-extensions:
+  org.freedesktop.Platform.ffmpeg-full:
+    directory: lib/ffmpeg
+    version: '20.08'
```
Comment (meisenzahl) [inline io.elementary.videos.yml:24]:
```suggestion
    version: '21.08'
```

Is there a reason why you refer to the outdated version here?

Comment (Marukesu) [reply io.elementary.videos.yml:24]: The 21.08 version doesn't work with the 6 runtime, once videos is updated to use the 6.1 runtime this should be updated to 21.08

Comment (meisenzahl) [reply io.elementary.videos.yml:24]: Thanks for clarification 🙂

**Comment by jeremypw** [comment]
Are there any licensing issues associated with this?

**Comment by Marukesu** [comment]
> Are there any licensing issues associated with this?

I believe there's no problem, since we aren't providing or downloading it.

The user still need to install it with `flatpak install freedesktop org.freedesktop.Platform.ffmpeg-full//20.08` or install an flatpak that uses the same version and auto-download it.

**Review by danrabbit — APPROVED** [review]
Since this just adds the extension point and doesn't download any codecs, I think we should be good here in terms of licensing

**Review by danrabbit — APPROVED** [review]

**Comment by Marine0705** [comment]
> Allows videos to use the codecs provided by the extension when installed.
> 
> Should fix #281.

Hello, I'm new to Linux and I have a problem launching videos. I found this solution but I don't know how to make it, can you explain it ? Thank you very much

**Pull request merged**